cfme/cloud/security_group.py

```diff
@@ -1,20 +1,268 @@
-""" Page functions for Security Group page
+# -*- coding: utf-8 -*-
+from navmazing import NavigateToAttribute, NavigateToSibling
+from widgetastic_manageiq import (
+    Accordion, BaseEntitiesView, BaseListEntity, BaseQuadIconEntity, BaseTileIconEntity,
+    BootstrapSelect, BreadCrumb, ItemsToolBarViewSelector, JSBaseEntity, ManageIQTree,
+    NonJSBaseEntity, ParametrizedView, SummaryTable, Table, Text, TextInput, Version, VersionPick
+)
+from widgetastic.widget import View, Select
+from widgetastic_patternfly import Dropdown, Button
+from cfme.base.ui import BaseLoggedInPage
+from cfme.exceptions import ItemNotFound, SecurityGroupsNotFound
+from cfme.web_ui import match_location
+from utils.appliance.implementations.ui import navigate_to, navigator, CFMENavigateStep
+from utils.appliance import Navigatable
+from utils.wait import wait_for
 
 
-:var list_page: A :py:class:`cfme.web_ui.Region` object describing elements on the list page.
-:var details_page: A :py:class:`cfme.web_ui.Region` object describing elements on the detail page.
-"""
+class SecurityGroupsToolbar(View):
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
 
-from cfme.web_ui import Region, SplitTable
+    view_selector = View.nested(ItemsToolBarViewSelector)
 
 
-# Page specific locators
-list_page = Region(
-    locators={
-        'security_group_table': SplitTable(header_data=('//div[@class="xhdr"]/table/tbody', 1),
-            body_data=('//div[@class="objbox"]/table/tbody', 1))
-    },
-    title='Security Groups')
+class SecurityGroupsDetailsToolbar(View):
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Button(title='Download summary in PDF format')
 
 
-details_page = Region(infoblock_type='detail')
+class SecurityGroupsDetailsAccordion(View):
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class SecurityGroupsQuadIconEntity(BaseQuadIconEntity):
+    pass
+
+
+class SecurityGroupsTileIconEntity(BaseTileIconEntity):
+    quad_icon = ParametrizedView.nested(SecurityGroupsQuadIconEntity)
+
+
+class SecurityGroupsListEntity(BaseListEntity):
+    pass
+
+
+class NonJSSecurityGroupsEntity(NonJSBaseEntity):
+    quad_entity = SecurityGroupsQuadIconEntity
+    list_entity = SecurityGroupsListEntity
+    tile_entity = SecurityGroupsTileIconEntity
+
+
+def SecurityGroupsEntity():  # noqa
```

Comment:
I don't think this is needed here since the classes aren't actually modifying the base class behavior. Please test with including (not nesting!) BaseEntitiesView from widgetastic_manageiq.